mpas_analysis/ocean/plot_transect_subtask.py

```diff
@@ -0,0 +1,407 @@
+        if self.plotObs:
+            verticalComparisonGridName = \
+                self.remapMpasClimatologySubtask.verticalComparisonGridName
+            remappedFileName = \
+                self.remapMpasClimatologySubtask.obsDatasets.get_out_file_name(
+                    transectName,
+                    verticalComparisonGridName)
+            remappedRefClimatology = xr.open_dataset(remappedFileName)
+        elif self.refConfig is not None:
+            climatologyName = self.remapMpasClimatologySubtask.climatologyName
+            remappedFileName = \
+                get_remapped_mpas_climatology_file_name(
+                    self.refConfig, season=season,
+                    componentName=self.componentName,
+                    climatologyName=climatologyName,
+                    comparisonGridName=transectName)
+            remappedRefClimatology = xr.open_dataset(remappedFileName)
+            refStartYear = self.refConfig.getint('climatology', 'startYear')
+            refEndYear = self.refConfig.getint('climatology', 'endYear')
+            if refStartYear != self.startYear or refEndYear != self.endYear:
+                self.refTitleLabel = '{}\n(years {:04d}-{:04d})'.format(
+                        self.refTitleLabel, refStartYear, refEndYear)
+
+        else:
+            remappedRefClimatology = None
+
+        self._plot_transect(remappedModelClimatology, remappedRefClimatology)
+        # }}}
+
+    def _plot_transect(self, remappedModelClimatology, remappedRefClimatology):
+        # {{{
+        """ plotting the transect """
+
+        season = self.season
+        config = self.config
+        configSectionName = self.configSectionName
+
+        mainRunName = config.get('runs', 'mainRunName')
+
+        x = remappedModelClimatology['x'].values
+        z = remappedModelClimatology['z'].values
+        nPoints = remappedModelClimatology.sizes['nPoints']
+        nz = remappedModelClimatology.sizes['nz']
+
+        if len(x.shape) == 1:
+            x = numpy.tile(x.reshape((len(x), 1)), (1, nz))
+
+        if len(z.shape) == 1:
+            z = numpy.tile(z.reshape((1, len(z))), (nPoints, 1))
+
+        modelOutput = nans_to_numpy_mask(
+            remappedModelClimatology[self.mpasFieldName].values)
+
+        if remappedRefClimatology is None:
+            refOutput = None
+            bias = None
+        else:
+            refOutput = nans_to_numpy_mask(
+                remappedRefClimatology[self.refFieldName].values)
+
+            bias = modelOutput - refOutput
+
+        filePrefix = self.filePrefix
+        outFileName = '{}/{}.png'.format(self.plotsDirectory, filePrefix)
+        title = '{} ({}, years {:04d}-{:04d})'.format(
+                self.fieldNameInTitle, season, self.startYear,
+                self.endYear)
+
+        # make a call to plot the transect here!
+
+        # This is just a placeholder for testing
+        plt.figure(figsize=(8, 13))
```

Comment:
@gstreletz, as the comment says, the code below is just a placeholder that produces a crappy figure for testing purposes.  Here's where I think you should instead call a function from the shared plotting routines to plot the transect vs. observations.  If you can make use of one of our existing plotting routines, that's great, though this would still take some modification since I don't think we currently have one that supports the 3-panel plot we would need for this purpose.  If you need to write a new one (and I suspect that might be the case), please do so.

I strongly suggest you try to keep things fairly general:
* use the existing infrastructure for making a color map and color bar based on the information from the `configSectionName` supplied to this task
* use titles of plots, units, etc. passed in to this task
* use font sizes, dpi, etc. provided by the config options similar to other tasks.

Feel free to push commits to this branch.